Atangster/130 update TTL and sharing methods for MovieList sharing

atangster opened the pull request:
## Pull Request Details

Link to associated issue: https://github.com/anusii/moviestar/issues/130

# Checklist
Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (make prep or flutter analyze lib)
- [x] Tested on at least one device
  - [x] Windows
  - [x] Linux
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

atangster committed:
update TTL and sharing methods for MovieList sharing

atangster committed:
Merge branch 'atangster/140_fix_import' into atangster/130_movielist_data

atangster committed:
Merge branch 'atangster/127_fix_lists' into atangster/130_movielist_data

anushkavidanage commented:
Not sure if the below is relevant to this issue. But when I share a movie, I cannot see the Owner and Shared by details of the movie in the shared movies list.

<img width="1265" height="841" alt="sharing-a-movie" src="https://github.com/user-attachments/assets/eeef2ed3-8e17-4825-b5d5-280ef75b530a" />

cdawei commented:
It may be good to first merge the current `dev` into this PR. I got the old problem as described here https://github.com/anusii/moviestar/pull/119#issuecomment-3111944768 which was fixed by the recently merged PR https://github.com/anusii/moviestar/pull/120.

cdawei reviewed (commented):
The code looks good to me. Thanks @atangster. It does not seem there's any user end changes to test until https://github.com/anusii/moviestar/issues/131 is implemented (please let me if this is not the case).

A few suggestions regarding the implementation:
- Instead of returning a `Map<String, dynamic>` in function `getSharedLists()`, it may be good to define a custom data structure (class) as the return type. This is in the same spirit as https://github.com/anusii/solidpod/issues/314. Similarly in function `getMySharedLists()`
- Please consider defining an enumeration of the operations used in function `canUserPerformOperation()`.

Noted that I still need to merge branch `dev` into this PR in order to retrieve the movies from TMDB.

atangster committed:
extend sharing for movielist and add ui

fix sharing and fetching of individual movie files in movie list

fix display of shared webid

remove debug logs

atangster committed:
fix comment format

atangster committed:
update solidpod dependency

atangster committed:
Fix lint

atangster committed:
Merge remote-tracking branch 'origin/dev' into atangster/132_extend_sharing

atangster committed:
Merge remote-tracking branch 'origin/dev' into atangster/130_movielist_data

atangster committed:
use custom data structure and enum

atangster committed:
format code

atangster committed:
Merge branch 'atangster/130_movielist_data' into atangster/132_extend_sharing

atangster committed:
update with custom data structure and enum

atangster commented:
hmm it appears running `dart format` made a lot of little formatting changes

the only really relevant changes to review are
- `lib/models/movie_list_operation.dart`
- `lib/models/shared_movie_list.dart`
- `lib/screens/search_screen.dart`

atangster committed:
run dart format

atangster committed:
fix import order

atangster committed:
Implement batch sharing UI for MovieStar collections

atangster committed:
Fix flutter analyze issues and apply code formatting

fix import order lint

atangster committed:
fix flutter analyze

atangster committed:
fix ux by handling movie file sharing under the hood

atangster committed:
fix code format

atangster committed:
add blank line between comment and code

atangster committed:
add custom list functionality

atangster committed:
fix code format

atangster committed:
fix flutter analyze

atangster committed:
add sharing functionality for custom lists

atangster committed:
format code

atangster committed:
run dart format

tonypioneer committed:
Fix typos

gjwgit committed:
Merge pull request #150 from anusii/atangster/134_custom_list

Atangster/134 add custom list functionality

gjwgit committed:
Merge pull request #152 from anusii/atangster/151_share_custom_list

Atangster/151 add sharing functionality for custom lists

gjwgit committed:
Merge pull request #145 from anusii/atangster/132_extend_sharing

Atangster/132 extend sharing for movielist and add ui

gjwgit reviewed (changes requested):
Tested to work with sub tasks merged.

Needs a dev merge and then tested again.

atangster committed:
Merge remote-tracking branch 'origin/dev' into atangster/130_movielist_data

atangster committed:
fix my lists and my movies tab after merge

atangster committed:
fix flutter analyze

atangster committed:
fix flutter analyze

atangster committed:
run dart format and import order lint

gjwgit reviewed (approved):
Tested. Working well.
